### PR TITLE
[PRTL-2592] save survival plots

### PR DESCRIPTION
**Pull request by samrichca**
## Ticket Number

PRTL-2592

## Environment to be used in testing

- [ ] `prod`
- [x] `dev-oicr`
- [ ] `qa` (which version?)

### CODE REVIEW

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js [hunk 1]
```diff
@@ -300,8 +313,7 @@ const ClinicalVariableCard = ({
         </Row>
       </Row>
       {isEmpty(tableData)
-        ? (
-          <Row
+        ? (<Row
```
Comment (caravinci) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js]: what is this changing for?

Comment (caravinci) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js]:
```suggestion
        ? (
          <Row
```

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js [hunk 2]
```diff
@@ -310,8 +322,7 @@ const ClinicalVariableCard = ({
             }}
             >
             There is no data for this facet
-          </Row>
-        )
+          </Row>)
```
Comment (caravinci) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js]:
```suggestion
          </Row>
        )
```

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js [hunk 3]
```diff
@@ -395,15 +406,27 @@ const ClinicalVariableCard = ({
...
-                    ? 'clinicalOverall'
-                    : 'categorical'}
-                  survivalData={selectedSurvivalBins.length === 0
-                    ? overallSurvivalData
-                    : selectedSurvivalData}
-                  survivalPlotLoading={survivalPlotLoading}
-                  />
+                variable.isSurvivalCustom &&
+                  selectedSurvivalBins.length === 0
+                    ? (<Row
```
Comment (caravinci) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js]:
```suggestion
                    ? (
                      <Row
```

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js [hunk 4]
```diff
@@ -395,15 +406,27 @@ const ClinicalVariableCard = ({
...
+                        id={`${wrapperId}-container`}
+                        style={{
+                          alignItems: 'center',
+                          flex: 1,
+                          justifyContent: 'center',
+                        }}
+                        >
+                        No matches found
+                      </Row>)
```
Comment (caravinci) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js]:
```suggestion
                      </Row>
                    )
```

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js [hunk 5]
```diff
@@ -395,15 +406,27 @@ const ClinicalVariableCard = ({
...
+                    : <ClinicalSurvivalPlot
```
Comment (caravinci) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js]:
```suggestion
                    : (
                      <ClinicalSurvivalPlot
```

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js [hunk 6]
```diff
@@ -395,15 +406,27 @@ const ClinicalVariableCard = ({
...
+                        plotType={selectedSurvivalBins.length === 0
+                          ? 'clinicalOverall'
+                          : 'categorical'}
+                        survivalData={selectedSurvivalBins.length === 0
+                          ? overallSurvivalData
+                          : selectedSurvivalData}
+                        survivalPlotLoading={survivalPlotLoading}
+                        />
```
Comment (caravinci) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js]:
```suggestion
                        />
                      )
```

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/CategoricalVariableCard.js [hunk 7]
```diff
@@ -137,8 +172,10 @@ export default compose(
       binsAreCustom,
       dataBuckets,
       dispatchUpdateClinicalVariable,
+      variable: { isSurvivalCustom }
     }) => ({
       resetBins: () => {
+        console.log('reset bins');
```
Comment (anncatton) [inline src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/CategoricalVariableCard.js]: i tested without this log statement, the function runs on my local

File: src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js [hunk 8]
```diff
@@ -81,84 +84,94 @@ const getTableData = ({
...
-          <Button
-            disabled={
-              bin.key === '_missing' ||
-              bin.chart_doc_count < MINIMUM_CASES ||
-              (selectedSurvivalBins.length >= MAXIMUM_CURVES &&
-                selectedSurvivalBins.indexOf(bin.key) === -1)
-            }
-            onClick={() => {
-              updateSelectedSurvivalBins(displayData, bin);
+}) => displayData.length === 0
```
[the remark this answers is not in the document]
Comment (caravinci) [reply src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/ClinicalVariableCard.js]: `displayData == []` at that point. 
but yes, the check is redundant. 

`[].map` is inexpensive enough to not require the check, as @yuanhengzhao pointed out the other day.

**Pull request merged**